### PR TITLE
link missing task or flow when merging configs

### PR DESCRIPTION
**Pull request by Br4nd0R**
# Changes
* Fixed an issue where CumulusCI was incorrectly merging task option overrides.

### QA Notes
I used the following in the standard lib while testing:
```
official_testing_flow:
        group: foo
        description: Does some stuff
        steps:
            1:
                task: util_sleep
            2:
                task: util_sleep
            3:
                task: util_sleep
   ```
   Then I modified the project `cumulusci.yml` to include:
   ```
   flows:
    official_testing_flow:
        steps:
            1:
                # task: util_sleep
                options:
                    seconds: 1
```
* Also verified setting the flow step to a value of `None` still works.

### CODE REVIEW

File: cumulusci/core/tests/test_utils_merge_config.py [hunk 1]
```diff
@@ -260,3 +262,15 @@ def test_cleanup_flow_step_override_conflicts__obsolete_flow_with_tasks(
     clean_configs = utils.cleanup_flow_step_override_conflicts(configs)
 
     assert expected_project_config == clean_configs["project_config"]
+
+
+def test_link_missing_task_or_flow():
```
Comment (prescod) [inline cumulusci/core/tests/test_utils_merge_config.py:267]: I'm not really a fan of this kind of testing. I think the automated tests should do what you described in the PR cmments. As a maintainer I don't care much that the new function `link_missing_task_or_flow` does the right thing. I care that CCI configs merge properly. I'd like to see some CCI configs merge. Then if I change the underlying implementation later, I'll still know that CCI configs merge properly.

For example, here is how we test the CCI validating linter.

https://github.com/SFDO-Tooling/CumulusCI/blob/8d46f48da6850d4b979d7f36f01f4b20103ed236/cumulusci/utils/yaml/tests/test_cumulusci_yml.py#L97

Comment (prescod) [reply cumulusci/core/tests/test_utils_merge_config.py:267]: IMO: A good rule of thumb is whether there's a clear line between the WI ACs and the test cases.

File: cumulusci/core/utils.py [hunk 2]
```diff
@@ -239,10 +239,27 @@ def remove_overridden_flow_steps_in_config(
...
+def link_missing_task_or_flow(
+    step_config_to_override: dict, overriding_step_config: dict
+):
+    """If the incoming override does not have task/flow defined then inherit from
+    the flow step that we're overridding."""
+    if "flow" not in overriding_step_config and "task" not in overriding_step_config:
+        if "task" in step_config_to_override:
+            overriding_step_config["task"] = step_config_to_override["task"]
+        elif "flow" in step_config_to_override:
+            overriding_step_config["flow"] = step_config_to_override["flow"]
```
Comment (davidmreed) [inline cumulusci/core/utils.py:260]: 👍

File: cumulusci/core/tests/test_utils_merge_config.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/core/tests/test_utils_merge_config.py:267]: I agree that we should build new tests to catch this type of problem, and I will write a WI to do so (along with documenting this specific wrinkle of flow customization). However, I don't want to hold up @Br4nd0R's fix to this bug to do that, as there is serious production impact we need to cure.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**